install.sh

```diff
@@ -317,6 +317,17 @@ EOABORT
 )"
 fi
 
+UNAME_MACHINE="$(uname -m)"
+if [[ "$UNAME_MACHINE" == "arm64" ]]; then
+  abort "$(cat <<EOABORT
+Homebrew is not (yet) supported on ARM processors!
+Rerun the Homebrew installer under Rosetta 2.
```

Comment:
```suggestion
To rerun the Homebrew installer under Rosetta 2,
select your Terminal app in Finder and click Get Info.
Make sure that "Open using Rosetta" is checked.
Then quit and re-open your Terminal.
```

Reply:
@claui Does this force your Terminal to always open in Rosetta? I'm wondering if there are any nicer ways than this as it seems like this may not be what people want, I dunno. If not: no worries.

Reply:
Yes, it forces the Terminal to always open in Rosetta.

It may not always be what people want but I’d still prefer to recommend it than recommending to prepend `arch -x86_64`.
The latter may be more convenient but:
- it’s extremely easy to forget during daily use, and
- imho it’s also kind of awkward to describe.

Reply:
What if Rosetta is not yet installed (as it sounds like it isn't/won't be by default)?



> It may not always be what people want but I’d still prefer to recommend it than recommending to prepend `arch -x86_64`.

Doing this and then having something like a `rosettabrew` alias that runs `arch -x86_64 brew` feels like it'd be nice.

Reply:
> What if Rosetta is not yet installed

Maybe add a link that tells people how to install Rosetta upfront. Like:

```
If Rosetta is not installed, please follow the instructions at <url>
```

Reply:
Leaving as-is for now but will happily 👀 any PR to improve the text. Thanks for the help and testing @claui ❤️

Reply:
> Leaving as-is for now but will happily 👀 any PR to improve the text.

@MikeMcQuaid Fyi, https://github.com/Homebrew/brew/pull/9126 includes a few suggestions. I plan to apply those here once they’re merged there.